packages/language/src/language-server/types.ts

```diff
@@ -56,6 +56,10 @@ export function rangeToLSP(
   };
 }
 
+export function tokenToUri(token: IToken): string {
+  return token.payload.uri._formatted; // TODO: Is this safe?
+}
```

Comment:
Suggestion: You can just use `toString()`. `_formatted` is just the cache for the `toString()` method inside of [`Uri`](https://github.com/microsoft/vscode-uri/blob/edfdccd976efaf4bb8fdeca87e97c47257721729/src/uri.ts#L410-L420). Note that as of my latest change, the URI is now optional and might return `undefined`.
```suggestion
export function tokenToUri(token: IToken): string | undefined {
  return token.payload?.uri?.toString();
}
```

Reply:
Ok! Should I just pass along an empty URI to the acceptor then?

Reply:
Yep that's acceptable. When we promote this up to a Diagnostic we just drop the ones that lack an associated URI.